### PR TITLE
docs(samples): add usage samples to show handling of LRO response Operation

### LINKED ISSUE #3: Container: example(s) for long-running operations
Please provide (reference) an example for converting `container_v1.types.Operation` into `api_core.operation.Operation`.

[`create_cluster`](https://googlecloudplatform.github.io/google-cloud-python/latest/_modules/google/cloud/container_v1/gapic/cluster_manager_client.html#ClusterManagerClient.create_cluster) (and other methods in the Cloud Client Library) return (long-running) `Operation`(s).

I assumed (hoped) this would be `google.api_core.operation.Operation` that I could `operation.result()` but, the method turns [`google.cloud.container_v1.types.Operation`](https://googlecloudplatform.github.io/google-cloud-python/latest/container/gapic/v1/types.html#google.cloud.container_v1.types.Operation)

[Spanner's Python Library, for example, [returns](https://googlecloudplatform.github.io/google-cloud-python/latest/spanner/instance-api.html#google.cloud.spanner_v1.instance.Instance.create) google.api_core.operation.Operation in this case]

I'm struggling to connect this with (presumably?) [Long-Running Operations](https://googlecloudplatform.github.io/google-cloud-python/latest/core/operation.html) documented in Core.

I'm unable to new an `OperationsClient`. The Example suggests `operations_v1.OperationsClient()` but this errors with `TypeError: __init__() takes at least 2 arguments (1 given).`

It's unclear what `channel (grpc.Channel)` I should provide.

Is this the correct approach? Having triggered the cluster creation, I'd like to track its progress to completion.

### PR DESCRIPTION
**Pull request by Shabirmean**
### Description
- For a detailed description read internal ref [b/189985582](http://b/189985582)
- The Kubernetes engine API protos were written before the API Improvement Proposal [(AIP-151)](https://google.aip.dev/151)
- AIP-151 suggests to annotate API calls that takes a long time to complete with a certain annotation. This will enable the auto-generator _(that generates our client libraries)_ to wrap the response an object of type [google.longrunning.Operation](https://cloud.google.com/service-infrastructure/docs/service-management/reference/rpc/google.longrunning) in a language native (Python Future or Nodejs Promise) async construct.
- However, since  AIP-151 came long after the container.v1 API protos were written, the container API protos don't have the necessary annotation as guided in AIP-151  
- As a result the auto generated **container client libraries** do not return the very useful [google.longrunning.Operation](https://cloud.google.com/service-infrastructure/docs/service-management/reference/rpc/google.longrunning) object, instead they return [google.container.v1.Operation](https://cloud.google.com/nodejs/docs/reference/container/latest/container/protos.google.container.v1.operation-class)
- This object does not have the necessary built-ins to easily verify if the operation has completed. 
- Thus this PR adds a few samples to show to customers how they can use this Operation object to validate if their operation has successfully completed using **polling with backoff on the getOperation()** API

### Changes
- Added 3 samples to showcase usage of the library and to show how to handle responses of type `google.container.v1.Operation`
    - A quickstart sample which simply lists the clusters
    - A create cluster and delete cluster example that returns an `Operation` object since they are long running operations
- Updated the READMEs explain the samples
- I have also added test cases for the added samples

### Testing
- Create a project
- Enable the container (GKE) API
- Create a service account with permissions to create/delete clusters _(can use Owner)_
- Download a key for the created service account

```bash
export GOOGLE_APPLICATION_CREDENTIALS="<PATH_TO_KEY>"
gcloud config set project <GCP_PROJECT>

cd <repo_path>
git checkout lro-fix
cd samples/snippets

python quickstart.js <GCP_PROJECT> us-central1-b
python create_cluster.js <GCP_PROJECT> us-central1-b gke10
python delete_cluster.js <GCP_PROJECT> us-central1-b gke10

nox -s py-3.6
```

### CODE REVIEW

**Comment by Shabirmean** [comment]
@busunkim96 - adding you in here since you had been part of the discussion in #3 and also [b/189985582](http://b/189985582). Can you TAL and see if it sufficiently addresses the requirements?

**Comment by Shabirmean** [comment]
@parthea - I've addressed the comments we discussed offline:
- Remove inline code for the landing page README
- Add region tags
- Add tests

Please TAL look when you have some time! Thank you 🙌🏼

File: samples/snippets/create_cluster_test.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import os
```
[the remark this answers is not in the document]
Comment (Shabirmean) [reply samples/snippets/create_cluster_test.py:16]: Added in https://github.com/googleapis/python-container/pull/191/commits/588e02c518855f97059c919446d66157d9fb0202

File: samples/snippets/create_cluster.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    parser.add_argument("project_id", help="Google Cloud project ID")
+    parser.add_argument("zone", help="GKE Cluster zone")
+    parser.add_argument("cluster_name", help="Name to be given to the GKE Cluster")
+    args = parser.parse_args()
+
+    if len(sys.argv) != 4:
+        parser.print_usage()
+        sys.exit(1)
+
+    create_cluster(args.project_id, args.zone, args.cluster_name)
```
[the remark this answers is not in the document]
Comment (Shabirmean) [reply samples/snippets/create_cluster.py:111]: I added this following this from some other **python-xx** samples. Yup, agree that it is not required. But added it as a helpful measure for a dev who want to quickly run and test it.

File: samples/snippets/create_cluster.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    cluster_def = {
+        "name": cluster_name,
+        "initial_node_count": 2,
+        "node_config": {"machine_type": "e2-standard-2"},
+    }
+    # Create the request object with the location identifier.
+    request = {"parent": cluster_location, "cluster": cluster_def}
+    create_response = client.create_cluster(request)
+    op_identifier = f"{cluster_location}/operations/{create_response.name}"
+    # poll for the operation status and schedule a retry until the cluster is created
```
[the remark this answers is not in the document]
Comment (Shabirmean) [reply samples/snippets/create_cluster.py:92]: It really depends on:
1. The operation itself _(Create cluster, delete cluster, autoscale cluster, etc)_
2. Various parameters that are used for the operation _(# of nodes in the cluster, other controllers etc)_

For the example used here it takes roughly ~300 seconds which is 5 minutes. With the backoff strategy used that is like 10-12 retries. I have kept the retry limit to 20 as a healthy upper bound; The polling stops after that many retries

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**